.github/workflows/go.yml

```diff
@@ -43,7 +43,7 @@ jobs:
         run: make api && git diff --exit-code
 
       - name: Run tests
-        run: go test -race -vet=off ./...
+        run: go test -race -vet=off $(go list ./... | grep -v github.com/bitcoin-sv/arc/test)
```

Comment:
I think the better solution to have separate execution of tests would be to run the e2e tests only if a certain env var is given e.g. `E2E_TESTS=1 go test -race -vet=off` as described here: https://konradreiche.com/blog/how-to-separate-integration-tests-in-go

Reply:
Either approach is fine.  I don't see any additional benefits of the aporoach you mention here. But open for suggestions as always.

Reply:
With this approach you used here this will always fail `go test ./...` unless I add this tail `$(go list ./... | grep -v github.com/bitcoin-sv/arc/test)`.

I'd rather have a solution where I have to add something if I want to run the e2e tests. Usually you'll not want to run them because they only make sense together with the respective docker-compose file

Reply:
Still running `go test ./...` would fail now. I prefer to have to add something like an env var in order to run the e2e tests over having to prevent running the e2e tests each time with this tail `$(go list ./... | grep -v github.com/bitcoin-sv/arc/test)`.
This commit shows how it can be done: https://github.com/bitcoin-sv/arc/commit/8c16a77f5a8d30dc6657802a03e348e21ed8a778

Reply:
it's env variable vs folder exclusion in go test. I don't see any exceptional benefit but will do it the way you insist.